schemas/json/.wp-env.json

```
@@ -0,0 +1,92 @@
+{
+	"title": "JSON schema for WordPress wp-env configuration files",
+	"$schema": "http://json-schema.org/draft-04/schema#",
+	"definitions": {
+		"//": {
+			"reference": "https://developer.wordpress.org/block-editor/reference-guides/packages/packages-env/"
+		},
+		"wpEnvProperties": {
+			"properties": {
+				"core": {
+					"description": "The WordPress installation to use. If null is specified, wp-env will use the latest production release of WordPress.",
+					"default": "null",
+					"oneOf": [
+						{
+							"type": "string"
+						},
+						{
+							"type": "null"
+						}
+					]
+				},
+				"phpVersion": {
+					"description": "The PHP version to use. If null is specified, wp-env will use the default version used with production release of WordPress.",
+					"default": "null",
+					"oneOf": [
+						{
+							"type": "string"
+						},
+						{
+							"type": "null"
+						}
+					]
+				},
+				"plugins": {
+					"description": "A list of plugins to install and activate in the environment.",
+					"default": [],
+					"type": "array",
+					"items": {
+						"type": "string"
+					}
+				},
+				"themes": {
+					"description": "A list of themes to install in the environment.",
+					"default": [],
+					"type": "array",
+					"items": {
+						"type": "string"
+					}
+				},
+				"port": {
+					"description": "The primary port number to use for the installation. You'll access the instance through the port: 'http://localhost:8888'.",
+					"default": 8888,
+					"type": "integer"
+				},
+				"config": {
+					"description": "Mapping of wp-config.php constants to their desired values.",
+					"default": {},
+					"type": "object"
+				},
+				"mappings": {
+					"description": "Mapping of WordPress directories to local directories to be mounted in the WordPress instance.",
+					"default": {},
+					"type": "object"
+				}
+			}
+		}
+	},
+	"type": "object",
+	"anyOf": [
+		{
+			"$ref": "#/definitions/wpEnvProperties"
+		},
+		{
+			"properties": {
+				"$schema": {
+					"type": "string"
+				},
+				"env": {
+					"description": "The key env is available to override any of the above options on an individual-environment basis.",
+					"type": "object",
+					"default": {},
+					"patternProperties": {
+						"[a-zA-Z]": {
+							"type": "object",
+							"$ref": "#/definitions/wpEnvProperties"
+						}
+					}
+				}
+			}
+		}
```

Setting `"additionalProperties": false` is kinda awkward when you're "merging" properties with `allOf`. You have to create an entirely new subschema that defines all the merged properties for it to work. You can exclude the contents of the property checks here because those are already tested earlier.

```suggestion
		},
		{
			"properties": {
				"core": {},
				"phpVersion": {},
				"plugins": {},
				"themes": {},
				"port": {},
				"config": {},
				"mappings": {},
				"env": {}
			},
			"additionalProperties": false
		}
```